GoogleSignIn/Tests/Unit/GIDVerifiedAccountDetailResultTest.m

```diff
@@ -0,0 +1,113 @@
+
+#import "GoogleSignIn/Sources/GIDVerifyAccountDetail/Fake/GIDVerifiedAccountDetailHandlingFake.h"
+
+#import "GoogleSignIn/Sources/GIDSignIn_Private.h"
+
+#import "GoogleSignIn/Tests/Unit/OIDAuthState+Testing.h"
+#import "GoogleSignIn/Tests/Unit/OIDAuthorizationRequest+Testing.h"
+#import "GoogleSignIn/Tests/Unit/OIDTokenResponse+Testing.h"
+
+@interface GIDVerifiedAccountDetailResultTest : XCTestCase
+@end
+
+@implementation GIDVerifiedAccountDetailResultTest : XCTestCase
+
+- (void)testInit {
+  OIDAuthState *authState = [OIDAuthState testInstance];
+
+  GIDVerifiableAccountDetail *verifiedAccountDetail =
+      [[GIDVerifiableAccountDetail alloc] initWithAccountDetailType:GIDAccountDetailTypeAgeOver18];
+
+  NSArray<GIDVerifiableAccountDetail *> *verifiedList =
+      @[verifiedAccountDetail, verifiedAccountDetail];
+
+  GIDVerifiedAccountDetailResult *result = 
+      [[GIDVerifiedAccountDetailResult alloc] initWithLastTokenResponse:authState.lastTokenResponse
+                                                         accountDetails:verifiedList
+                                                              authState:authState];
+
+  XCTAssertEqual(result.verifiedAuthState, authState);
+  XCTAssertEqual(result.verifiedAccountDetails, verifiedList);
+  XCTAssertEqual(result.expirationDate, authState.lastTokenResponse.accessTokenExpirationDate);
+  XCTAssertEqual(result.accessTokenString, authState.lastTokenResponse.accessToken);
+  XCTAssertEqual(result.refreshTokenString, authState.lastTokenResponse.refreshToken);
+
+}
+
+- (void)testRefreshTokensWithCompletion_success {
+  GIDVerifiableAccountDetail *verifiedAccountDetail = 
+      [[GIDVerifiableAccountDetail alloc] initWithAccountDetailType:GIDAccountDetailTypeAgeOver18];
+
+  NSString *kAccountDetailList = [NSString stringWithFormat:@"%@ %@",
+                                  kAccountDetailTypeAgeOver18Scope,
+                                  kAccountDetailTypeAgeOver18Scope];
+  OIDTokenResponse *tokenResponse = [OIDTokenResponse testInstanceWithScope:kAccountDetailList];
+  OIDAuthState *authState = [OIDAuthState testInstanceWithTokenResponse:tokenResponse];
+  GIDVerifiedAccountDetailHandlingFake *result =
+      [[GIDVerifiedAccountDetailHandlingFake alloc] initWithTokenResponse:authState.lastTokenResponse
+                                                        verifiedAuthState:authState
+                                                                    error:nil];
+
+  NSArray<GIDVerifiableAccountDetail *> *expectedVerifiedList =
+      @[verifiedAccountDetail, verifiedAccountDetail];
+
+  XCTestExpectation *expectation = [self expectationWithDescription:@"Completion called"];
+  [result refreshTokensWithCompletion:^(GIDVerifiedAccountDetailResult * _Nullable refreshedResult,
+                                      NSError * _Nullable error) {
+    XCTAssertNil(error);
+    XCTAssertNotNil(refreshedResult);
```

Comment:
Are you able to assert equality on this `refreshedResult` and some `expectedVerifiedAccountDetailResult`?

Reply:
Done. Could not use `XCTAssertEqualObjects` so I decided to override the `isEqual:` method and check for all the properties.